src/query.ts

```diff
@@ -204,231 +243,174 @@ export class Query {
+      this.transformer(this.params['transformer']);
+      delete this.params['transformer'];
+    }
+    if ('type' in this.params) {
+      this.addTypes(this.params['type']);
+      delete this.params['type'];
+    }
+    if ('nostats' in this.params) {
+      this.noStats(this.params['nostats']);
+      delete this.params['nostats'];
+    }
+    if ('query_name' in this.params) {
+      this.queryName(this.params['query_name']);
+      delete this.params['query_name'];
+    }
+    if ('timeout' in this.params) {
+      this.timeout(this.params['timeout']);
+      delete this.params['timeout'];
+    }
+    if ('jsonp' in this.params) {
+      this.jsonp(this.params['jsonp']);
+      delete this.params['jsonp'];
     }
   }
 
-  /**
-   * Adds or alters a sorting parameter to the query
-   *
-   * @param  {String}   field   The field to order by
-   *
-   * @param  {String}   order   Order literal to order by
-   *
-   */
-  public addSorting(field: string, order = Sort.ASC): void {
-    // This is so Prettier won't change it to const automagically
-    let sortParams: RequestSortOptions = [];
-    sortParams = this.params.sort || [];
-    this.removeSorting(field);
-    const obj: GenericObject = {};
-    obj[field] = order;
-    (sortParams as Array<RequestSortOptions>).push(obj);
-
-    this.params.sort = sortParams;
-  }
-
-  /**
-   * Sets all the sorting options at once
-   *
-   * @param  {Array}  sortings    The sortings in the order wanted
-   *
-   */
-  public setSorting(sortings: Array<RequestSortOptions>): void {
-    this.params.sort = sortings as RequestSortOptions;
+  public addSort(fieldOrList: InputSortValue[]): Query;
+  public addSort(fieldOrList: string, sortType?: string): Query;
+  public addSort(fieldOrList: string | InputSortValue[], sortType: string = OrderType.ASC): Query {
+    if (typeof fieldOrList === 'string') {
+      this._sort.set(fieldOrList, this._validateOrderType(sortType));
+    } else {
+      for (const value of fieldOrList) {
+        if (typeof value === 'string') {
+          this._sort.set(value, OrderType.ASC);
+        } else {
+          const field = Object.keys(value)[0];
+          const sortType: string = Object.values(value)[0] as string;
+          this._sort.set(field, this._validateOrderType(sortType));
+        }
+      }
+    }
+    return this;
   }
 
   public hasSorting(field: string): boolean {
-    if (!this.params.sort) {
-      return false;
-    } else {
-      return this._findSortField(field, this.params.sort as Array<RequestSortOptions>) !== -1;
-    }
+    return this._sort.has(field);
   }
 
   /**
    * Sets the page value of the request, useful for pagination
    *
-   * @param  {Number}     page              The page we want to set
+   * @param  page - The page we want to set
+   *
+   * @returns Query
    *
    */
-  public page(page?: number): void {
-    if (page) {
-      this.setParameter('page', page);
+  public page(page?: number): Query {
```

Comment:
In this case I would reset page to `1` if the provided value is `0`, `undefined` or `null`. This is a parameter with a clear default value.

```typescript
const value: number = parseInt(page || 1, 10);
if (isNaN(value)) {
  throw new QueryValueError();
} else {
  this._page = value;
}
```